### DISCUSSION TITLE
Seeder in Craft core

### DISCUSSION THREAD

**Discussion by acalvino4** [craftcms/cms#10364] [Ideas]
## Background
The processes of automated testing, db version control, and site bootstrapping are greatly aided by Migration and Seeder systems/utilities. [Laravel](https://laravel.com/docs/8.x/seeding) (and most MVC frameworks, I believe) have this functionality baked it, and while it doesn't appear (from my cursory search) that yii has this in the core, there is an [extension](https://www.yiiframework.com/extension/yii2-db-seeder) for it.

Migrations are very effectively handled through project-config, which is just one of the many things that make Craft so awesome to work with. As far as I can tell though, there is no system or utility for Seeding in Craft. There was a plugin (no idea how good it was) that is now abandoned.

## Feature description
* Standardized file format & location for specifying seed data for Entries/Users/Categories/etc. ideally more toward the yaml end of things than php; that is, config-like rather than code-like
* Control panel interface for exporting created entries/users/etc. to config files that can be committed ("seeder sets")
* `craft seed` command for applying these changes (probably a way in the cp as well)
* Bonus ask: a faker library integration for generating dummy data

## Use Cases
* Provide an easy, standardized way of seeding the db with records for tests
* Provide way for Entries/Users/etc. to be part of a site's boilerplate - i.e. if an org finds themselves repeatedly creating certain records for every new site
* Export entries for copying prod to staging, or vice versa.

**Comment by timkelty** [comment]
Related discussion: https://github.com/craftcms/cms/discussions/9179

**Comment by acalvino4** [comment]
Related discussion: https://github.com/craftcms/cms/discussions/8953
